package/health_daemon/src/gnss_obs_monitor.c

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright (C) 2018 Swift Navigation Inc.
+ * Contact: Swift Navigation <dev@swiftnav.com>
+ *
+ * This source is subject to the license found in the file 'LICENSE' which must
+ * be be distributed together with this source. All other rights reserved.
+ *
+ * THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND,
+ * EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
+ */
+
+#include <stdlib.h>
+#include <string.h>
+
+#include <libpiksi/logging.h>
+#include <libpiksi/sbp_zmq_pubsub.h>
+
+#include <libsbp/sbp.h>
+#include <libsbp/observation.h>
+
+#include "health_monitor.h"
+#include "utils.h"
+
+#include "gnss_obs_monitor.h"
+
+/* these are from fw private, consider moving to libpiski */
+#define MSG_FORWARD_SENDER_ID (0u)
+
+#define GNSS_OBS_ALERT_RATE_LIMIT (10000u) /* ms */
+
+/* Below stolen from libswiftnav/signal.h */
+#define CODE_GLO_L1OF  (3u)
+#define CODE_GLO_L2OF  (4u)
+
+/** Is this a GLO signal?
+ *
+ * \param   code  Code to check.
+ * \return  True if this is a GLO signal.
+*/
+static inline bool is_glo(u8 code)
+{
+  return (CODE_GLO_L1OF == code) || (CODE_GLO_L2OF == code);
+}
+
+static health_monitor_t *gnss_obs_monitor;
+
+static struct gnss_obs_ctx_s {
+  bool gnss_read_resp;
+  bool gnss_enabled;
+} gnss_obs_ctx = {
+  .gnss_read_resp = false,
+  .gnss_enabled = false
+};
+
+static void sbp_msg_read_resp_callback(u16 sender_id, u8 len,
+                                       u8 msg_[], void *ctx)
+{
+  (void)sender_id;
+  (void)len;
+  health_monitor_t *monitor = (health_monitor_t *)ctx;
+
+  const char *section, *name, *value;
+  if(health_util_parse_setting_read_resp(msg_, len,
+                                         &section, &name, &value) == 0) {
+    bool last_gnss_enabled = gnss_obs_ctx.gnss_enabled;
+    if (health_util_check_gnss_enabled(section, name, value,
+                                       &gnss_obs_ctx.gnss_enabled) == 0) {
+      gnss_obs_ctx.gnss_read_resp = true;
+      if (gnss_obs_ctx.gnss_enabled && !last_gnss_enabled) {
+        health_monitor_reset_timer(monitor);
+      }
+    }
+  }
+}
+
+static bool check_obs_msg_for_gnss(u8 *msg, u8 len)
```

Comment:
How about `check_obs_msg_for_glo_obs`

Reply:
+1

Reply:
This comment is mostly to try and understand the need for the change, which I have no objection to, since I have seen glonass, gnss, glo throughout these repos:
 * libsbp ([glonass](https://github.com/swift-nav/libsbp/search?utf8=%E2%9C%93&q=glonass&type=), [gnss](https://github.com/swift-nav/libsbp/search?utf8=%E2%9C%93&q=gnss&type=), [glo](https://github.com/swift-nav/libsbp/search?utf8=%E2%9C%93&q=glo&type=))
 * piksi_firmware_private ([glonass](https://github.com/swift-nav/piksi_firmware_private/search?utf8=%E2%9C%93&q=glonass&type=), [gnss](https://github.com/swift-nav/piksi_firmware_private/search?utf8=%E2%9C%93&q=gnss&type=), [glo](https://github.com/swift-nav/piksi_firmware_private/search?utf8=%E2%9C%93&q=glo&type=))
 * libswiftnav-private ([glonass](https://github.com/swift-nav/libswiftnav-private/search?utf8=%E2%9C%93&q=glonass&type=), [gnss](https://github.com/swift-nav/libswiftnav-private/search?utf8=%E2%9C%93&q=gnss&type=), [glo](https://github.com/swift-nav/libswiftnav-private/search?utf8=%E2%9C%93&q=glo&type=))

If these sorts of naming issues come up is there a succinct way to understand when is appropriate to use which?

Reply:
More simply stated, when are glo/gnss/glonass not the same thing?